docker/chain/Dockerfile

```diff
@@ -1,63 +1,72 @@
-ARG expose_via=local
-
-FROM golang:1.22.3 AS base
+FROM docker:24.0.5-dind AS base
 
 ARG arch=amd64
 ARG cloudflare_token_http="not-a-token"
 ARG cloudflare_token_ws="not-a-token"
-ARG geth_version=v1.13.5
-
-WORKDIR /geth
-
-RUN git clone --quiet --branch ${geth_version} --depth 1 https://github.com/ethereum/go-ethereum .
-RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build go run build/ci.go install -static ./cmd/geth
-
-RUN /geth/build/bin/geth version
-RUN mv /geth/build/bin/geth /usr/local/bin/
-
-ARG admin_address="0x0"
-
-RUN touch fund-admin
-RUN echo "#!/bin/bash" >> fund-admin
-RUN echo "geth --exec \"eth.sendTransaction({from: eth.coinbase, to: \\\"${admin_address}\\\", value: new web3.BigNumber(eth.getBalance(eth.coinbase)).minus(web3.toWei(1, \\\"ether\\\"))  })\" attach /data/chain/geth.ipc" >> fund-admin
-RUN chmod +x fund-admin
-
-RUN touch run-node
-RUN echo "#!/bin/bash" >> run-node
-RUN echo "geth --datadir /data/chain --dev --ws --ws.api web3,eth,net --ws.addr 0.0.0.0 --ws.port 8546 --ws.origins '*' --http --http.api web3,eth,net --http.addr 0.0.0.0 --http.corsdomain '*' --http.port 8545 --http.vhosts '*' &" >> run-node
-RUN chmod +x run-node
-
-RUN touch reset
-RUN echo "#!/bin/bash" >> reset
-RUN echo "echo '- Kill geth'" >> reset
-RUN echo "pkill -INT geth" >> reset
-RUN echo "sleep 5" >> reset
-RUN echo "echo '- Clear data'" >> reset
-RUN echo "rm -rf /data/chain/*" >> reset
-RUN echo "echo '- Restart geth'" >> reset
-RUN echo "./run-node" >> reset
-RUN echo "sleep 5" >> reset
-RUN echo "echo '- Fund admin'" >> reset
-RUN echo "./fund-admin" >> reset
-RUN echo "echo '- Done'" >> reset
-RUN chmod +x reset
 
+RUN apk update
+RUN apk add bash
+RUN apk add nodejs
+RUN apk add npm
+
+WORKDIR /l2chain
+COPY . .
 RUN touch run
-RUN echo "#!/bin/bash" >> run
+RUN echo "dockerd-entrypoint.sh & " >> run
```

Comment:
Let's add a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) to run `!#/bin/bash`